Sources/Sentry/SentryScreenshot.m

```diff
@@ -0,0 +1,31 @@
+#import "SentryScreenShot.h"
+#import "SentryUIApplication.h"
+
+#if SENTRY_HAS_UIKIT
+#    import <UIKit/UIKit.h>
+
+@implementation SentryScreenshot
+
+- (NSArray<NSData *> *)appScreenshots
+{
+    NSArray<UIWindow *> *windows = [SentryUIApplication windows];
+
+    NSMutableArray *result = [NSMutableArray arrayWithCapacity:windows.count];
+
+    for (UIWindow *window in windows) {
+        UIGraphicsBeginImageContext(window.frame.size);
```

Comment:
`h`: The docs state that this function can be called from any thread. Are we sure this is true for all the logic in here? If so, can we please add a comment, that it is fine to call these methods from any thread?

Reply:
We need to be in the main thread. Fixing this.